src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -991,6 +998,327 @@ def progress_callback(current, total):
+    try:
+        if slot:
+            response = web_client.create_or_update_site_container_slot(resource_group, name,
+                                                                       slot, container_name, sitecontainer)
+        else:
+            response = web_client.create_or_update_site_container(resource_group, name, container_name, sitecontainer)
+        return response
+    except Exception as ex:
+        raise AzureInternalError("Failed to create or update sitecontainer {}. Error: {}"
+                                 .format(container_name, str(ex)))
+
+
+def update_webapp_sitecontainer(cmd, name, resource_group, container_name, slot=None, image=None, target_port=None,
+                                startup_cmd=None, is_main=None, system_assigned_identity=None,
+                                user_assigned_identity=None, registry_username=None, registry_password=None):
+    # get the sitecontainer
+    site_container = None
+    try:
+        site_container = get_webapp_sitecontainer(cmd, name, resource_group, container_name, slot)
+    except:
+        raise ResourceNotFoundError("Sitecontainer '{}' does not exist, failed to update the sitecontainer."
+                                    .format(container_name))
+    # update only the provided parameters
+    if image is not None:
+        site_container.image = image
+    if target_port is not None:
+        site_container.target_port = target_port
+    if startup_cmd is not None:
+        site_container.start_up_command = startup_cmd
+    if is_main is not None:
+        site_container.is_main = is_main
+    if system_assigned_identity is not None:
+        site_container.auth_type = AuthType.SYSTEM_IDENTITY
+    if user_assigned_identity is not None:
+        site_container.auth_type = AuthType.USER_ASSIGNED
+        site_container.user_managed_identity_client_id = user_assigned_identity
+    if registry_username is not None and registry_password is not None:
+        site_container.auth_type = AuthType.USER_CREDENTIALS
+        site_container.user_name = registry_username
+        site_container.password_secret = registry_password
+    response = _create_or_update_webapp_sitecontainer_internal(cmd, name, resource_group,
+                                                               container_name, site_container, slot)
+    return response
+
+
+def get_webapp_sitecontainer(cmd, name, resource_group, container_name, slot=None):
+    web_client = get_mgmt_service_client(cmd.cli_ctx, WebSiteManagementClient).web_apps
+    site_container = None
+    try:
+        if slot:
+            site_container = web_client.get_site_container_slot(resource_group, name, container_name, slot)
+        else:
+            site_container = web_client.get_site_container(resource_group, name, container_name)
+    except Exception as ex:
+        raise ResourceNotFoundError("Failed to fetch sitecontainer '{}'. Error: {}".format(container_name, str(ex)))
+    return site_container
```

Comment:
given site_container = None will not be returned in any case. could we just do 

try:
    if slot:
         return <>
     
     return <>
except:
     throw ?